### PR TITLE
feature/dashboards-redraw-event

### PR DESCRIPTION
**Pull request by sebastianbochan**
Added user-defined `redraw` event for dashboard components.

### CODE REVIEW

File: ts/Dashboards/Components/Component.ts [hunk 1]
```diff
@@ -730,6 +730,10 @@ abstract class Component {
             width,
             height
         });
+
+        fireEvent(this, 'redraw', {
+            component: this
+        });
```
Comment (pawellysy) [inline ts/Dashboards/Components/Component.ts:736]: IDK if this is accurate. it is added in the `resize` method. Apart from that, there already is a `redraw` event in the Component Class. 
https://github.com/highcharts/highcharts/blob/4181cfcd97f85e7de70fb514333434ce1c49f7b0/ts/Dashboards/Components/Component.ts#L966

**Review by pawellysy — CHANGES_REQUESTED** [review]

File: ts/Dashboards/Components/Component.ts (hunk 1 above)
Comment (karolkolodziej) [reply ts/Dashboards/Components/Component.ts:736]: Perhaps this should be changed?

**Review by pawellysy — APPROVED** [review]
So there is no changes in the core, just few tests added, right? If so, Looks good to me :D

File: test/cypress/integration/Dashboards/Components/events.cy.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        .trigger('mousemove', { clientX: 300 })
+        .trigger('mouseup');
+
+    cy.get('#resize').should('have.value', 'resize');
+  });
+
+  it('Update event should be triggered after sidebar update', () => {
+    cy.get('.highcharts-dashboards-component').first().click();
+    cy.get('.highcharts-dashboards-edit-toolbar-cell > .highcharts-dashboards-edit-toolbar-item:nth-child(2)').click();
+    cy.contains('Confirm').click();
```
Comment (karolkolodziej) [inline test/cypress/integration/Dashboards/Components/events.cy.js:47]: You forgot to check the update event.

**Review by karolkolodziej — APPROVED** [review]

**Review by karolkolodziej — APPROVED** [review]

File: test/cypress/integration/Dashboards/Components/events.cy.js (hunk 2 above)
Comment (sebastianbochan) [reply test/cypress/integration/Dashboards/Components/events.cy.js:47]: Good point!

**Pull request merged**